FindTBB.cmake is still not working

_Please use the following template to help us solving your issue._

## Issue Details

This is a follow-up of ticket #4236 which was closed. Since that ticket was closed, I have to create a new one. Hope it does not violate any federal or state law.

Now I download the raw version of FindTBB.cmake and overwrite the existing one under cmake\modules. The following are configure errors for demo/Polyhedron:

> -- Configuring demo in demo/Polyhedron
> Multi-configuration CMake generator: cannot display flags
> Requested component: Qt5
> Requested component: ImageIO
> Requested component: MPFR
> Requested component: GMP
> Found Eigen3: D:/3rd-parties/eigen-3.3.7 (found suitable version "3.3.7", minimum required is "3.2.0") 
> Can not locate METIS include directory
> Can not locate METIS library
> Could NOT find METIS (missing: METIS_LIBRARIES METIS_INCLUDE_DIRS) 
> **NOTICE: Intel TBB was not found. Bilateral smoothing and WLOP plugins are faster if TBB is linked.**
> NOTICE: This demo requires Qt5, and will not be compiled.

Am I safe to say FindTBB.cmake is still not working now? I have already set the `TBB_DIR` to the correct folder (`D:\3rd-parties\tbb2019_20190605oss` which contains CHANGES, LICENSE, README.md, bin, include, lib, etc). Thank you.

_Describe your issue. Please be specific (compilation error, runtime error, unexpected behavior, wrong results, etc.)._

## Source Code

https://github.com/CGAL/cgal/blob/master/Installation/cmake/modules/FindTBB.cmake
(note: I know this webpage contains HTML tags)

_If your issue arises by using CGAL in your own source code, please provide a minimalist example that we can compile easily to reproduce the bug. If your issue arises from using a CGAL program (demo, example, etc.), please let us know which one. Helping you solving an issue is much easier and efficient if we can reproduce it._

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Windows 10, version 1903, 64bit
* Compiler: Not compiled yet
* Release or debug mode: N/A
* Specific flags used (if any): N/A
* CGAL version: 4.14
* Boost version: it has nothing to do with Boost
* Other libraries versions if used (Eigen, TBB, etc.): TBB version 2019 Update 8.

If I (correctly) set TBB_ROOT, I got the following error:

> -- Configuring demo in demo/Triangulation_3
> CMake Warning (dev) at demo/Triangulation_3/CMakeLists.txt:45 (find_package):
>   Policy CMP0074 is not set: find_package uses <PackageName>_ROOT variables.
>   Run "cmake --help-policy CMP0074" for policy details.  Use the cmake_policy
>   command to set the policy and suppress this warning.
> 
>   CMake variable TBB_ROOT is set to:
> 
>     D:/3rd-parties/tbb2019_20190605oss
> 
>   For compatibility, CMake is ignoring the variable.
> This warning is for project developers.  Use -Wno-dev to suppress it.
> 
> CMake Error at demo/Triangulation_3/CMakeLists.txt:71 (CGAL_target_use_TBB):
>   Unknown CMake command "CGAL_target_use_TBB".
> 
> 
> Configuring incomplete, errors occurred!
> 

I am using the most-recent one downloaded from https://github.com/CGAL/cgal/blob/master/Installation/cmake/modules/FindTBB.cmake, so I think I should use `TBB_ROOT`. If I removed `TBB_ROOT` as well as `TBB_DIR`, I got the following error:

> -- Configuring demo in demo/Triangulation_3
> CMake Error at demo/Triangulation_3/CMakeLists.txt:71 (CGAL_target_use_TBB):
>   Unknown CMake command "CGAL_target_use_TBB".
>

[**@zhfsu**](https://github.com/zhfsu) commented on [Sep 26, 2019, 5:45 AM GMT+2](https://github.com/CGAL/cgal/issues/4240#issuecomment-535320227 "2019-09-26T03:45:55Z - Replied by Github Reply Comments"):
> If I (correctly) set TBB_ROOT, I got the following error:
> 
> > -- Configuring demo in demo/Triangulation_3

[skip warning]

> > CMake Error at demo/Triangulation_3/CMakeLists.txt:71 (CGAL_target_use_TBB):
> >   Unknown CMake command "CGAL_target_use_TBB".
> > 
> > 
> > Configuring incomplete, errors occurred!
> > 
> 
> I am using the most-recent one downloaded from https://github.com/CGAL/cgal/blob/master/Installation/cmake/modules/FindTBB.cmake, so I think I should use `TBB_ROOT`.

You are right. The issue here is that you have used only one part of https://github.com/CGAL/cgal/pull/4085, with one modified file. But that pull-request changed several files, and all the changes are needed.

I suggest you copy all the files from that pull-request, or use directly the Git branch `releases/CGAL-4.14-branch` from our Git repository.

--- a/Installation/cmake/modules/FindTBB.cmake
+++ b/Installation/cmake/modules/FindTBB.cmake
@@ -192,6 +192,11 @@ endmacro()
 # Get path, convert backslashes as ${ENV_${var}}
 getenv_path(TBB_ROOT)
 
+if(NOT ENV_TBB_ROOT)
+  getenv_path(TBBROOT)
+  set(ENV_TBB_ROOT ${ENV_TBBROOT})
+endif()
+
 # initialize search paths
 set(TBB_PREFIX_PATH ${TBB_ROOT} ${ENV_TBB_ROOT})
 set(TBB_INC_SEARCH_PATH "")

